nixos/modules/config/mysql.nix

```diff
@@ -0,0 +1,471 @@
+      };
+      pam = mkOption {
+        description = "Settings for <literal>pam_mysql</literal>";
+        type = types.submodule {
+          options = {
+            table = mkOption {
+              type = types.str;
+              example = "users";
+              description = "The name of table that maps unique login names to the passwords.";
+            };
+            updateTable = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              example = "users_updates";
+              description = ''
+                The name of the table used for password alteration. If not defined, the value
+                of the <literal>table</literal> option will be used instead.
+              '';
+            };
+            userColumn = mkOption {
+              type = types.str;
+              example = "username";
+              description = "The name of the column that contains a unix login name.";
+            };
+            passwordColumn = mkOption {
+              type = types.str;
+              example = "password";
+              description = "The name of the column that contains a (encrypted) password string.";
+            };
+            statusColumn = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              example = "status";
+              description = ''
+                The name of the column or an SQL expression that indicates the status of
+                the user. The status is expressed by the combination of two bitfields
+                shown below:
+
+                bit 0 (0x01): if flagged, <literal>pam_mysql</literal> deems the account to be expired and
+                              returns <literal>PAM_ACCT_EXPIRED</literal>. That is, the account is supposed
+                              to no longer be available. Note this doesn't mean that <literal>pam_mysql</literal>
+                              rejects further authentication operations.
+
+                bit 1 (0x02): if flagged, <literal>pam_mysql</literal> deems the authentication token
+                              (password) to be expired and returns <literal>PAM_NEW_AUTHTOK_REQD</literal>.
+                              This ends up requiring that the user enter a new password.
+              '';
+            };
+            passwordCrypt = mkOption {
+              example = "2";
+              type = types.enum [
+                "0" "plain"
+                "1" "Y"
+                "2" "mysql"
+                "3" "md5"
+                "4" "sha1"
+                "5" "drupal7"
+                "6" "joomla15"
+                "7" "ssha"
+                "8" "sha512"
+                "9" "sha256"
+              ];
+              description = ''
+                The method to encrypt the user's password:
+
+                0 (or "plain") 	= No encryption. Passwords are stored in plaintext.
```

Comment:
Indents like this won't have any effect in the resulting manpages or html manual.
It's a little bit more verbose, but I'd suggest to do it the docbook way:

```xml
<itemizedlist>
<listitem><para><literal>0</literal> (or <literal>"plain"</literal>): ...</para></listitem>
</itemizedlist>
```

Reply:
I have already checked how it will look in the manual and it looked okay-ish. But I will change it to the docbook way.

Reply:
Done.